presto-session-property-managers/src/main/java/io/prestosql/plugin/session/AbstractSessionPropertyManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.session;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.session.SessionConfigurationContext;
+import io.prestosql.spi.session.SessionPropertyConfigurationManager;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AbstractSessionPropertyManager
+        implements SessionPropertyConfigurationManager
+{
+    protected abstract List<SessionMatchSpec> getSessionMatchSpecs();
+
+    private Map<String, String> getSessionProperties(SessionConfigurationContext context)
+    {
+        // later properties override earlier properties
+        Map<String, String> combinedProperties = new HashMap<>();
+        for (SessionMatchSpec sessionMatchSpec : getSessionMatchSpecs()) {
+            combinedProperties.putAll(sessionMatchSpec.match(context));
+        }
+        return ImmutableMap.copyOf(combinedProperties);
+    }
+
+    @Override
+    public Map<String, String> getSystemSessionProperties(SessionConfigurationContext context)
+    {
+        Map<String, String> combinedSystemSessionProperties = getSessionProperties(context)
+                .entrySet()
+                .stream()
+                .filter(entry -> !entry.getKey().contains("."))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
collect to immutable map directly

Reply:
Done
Thanks! didn't knew about ImmutableMap collectors